rest_witchcraft/serializers.py

```diff
@@ -355,7 +359,9 @@ def get_fields(self):
 
             source = self._extra_kwargs.get(field_name, {}).get("source") or field_name
 
-            _fields[field_name] = self.build_field(source, info, self.model, depth)
+            _fields[field_name] = field = self.build_field(source, info, self.model, depth)
+            if self.partial_by_pk and source not in info.primary_keys:
```

Comment:
so this needs to only mark it as not required when field is in data. might move this to `to_internal_value` since here would have to inspect `self.root.initial_data`....